pandas/src/inference.pyx

```diff
@@ -928,161 +923,160 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
                 itimedeltas[i] = convert_to_timedelta64(val, 'ns')
-                seen_timedelta = 1
+                seen.stimedelta = 1
             else:
-                seen_object = 1
+                seen.sobject = 1
                 break
         elif util.is_integer_object(val):
-            seen_int = 1
+            seen.sint = 1
             floats[i] = <float64_t> val
             complexes[i] = <double complex> val
-            if not seen_null:
-                seen_uint = seen_uint or (int(val) > iINT64_MAX)
-                seen_sint = seen_sint or (val < 0)
+            if not seen.snull:
+                seen.saw_int(int(val))
 
-                if seen_uint and seen_sint:
-                    seen_object = 1
+                if seen.suint and seen.ssint:
+                    seen.sobject = 1
                     break
 
-                if seen_uint:
+                if seen.suint:
                     uints[i] = val
-                elif seen_sint:
+                elif seen.ssint:
                     ints[i] = val
                 else:
                     uints[i] = val
                     ints[i] = val
 
         elif util.is_complex_object(val):
             complexes[i] = val
-            seen_complex = 1
+            seen.scomplex = 1
         elif PyDateTime_Check(val) or util.is_datetime64_object(val):
 
             # if we have an tz's attached then return the objects
             if convert_datetime:
                 if getattr(val, 'tzinfo', None) is not None:
-                    seen_datetimetz = 1
+                    seen.sdatetimetz = 1
                     break
                 else:
-                    seen_datetime = 1
+                    seen.sdatetime = 1
                     idatetimes[i] = convert_to_tsobject(
                         val, None, None, 0, 0).value
             else:
-                seen_object = 1
+                seen.sobject = 1
                 break
         elif try_float and not util.is_string_object(val):
             # this will convert Decimal objects
             try:
                 floats[i] = float(val)
                 complexes[i] = complex(val)
-                seen_float = 1
+                seen.sfloat = 1
             except Exception:
-                seen_object = 1
+                seen.sobject = 1
                 break
         else:
-            seen_object = 1
+            seen.sobject = 1
             break
 
-    seen_numeric = seen_complex or seen_float or seen_int
+    seen.snumeric = seen.scomplex or seen.sfloat or seen.sint
 
     # we try to coerce datetime w/tz but must all have the same tz
-    if seen_datetimetz:
+    if seen.sdatetimetz:
         if len(set([getattr(val, 'tzinfo', None) for val in objects])) == 1:
             from pandas import DatetimeIndex
             return DatetimeIndex(objects)
-        seen_object = 1
-
-    if not seen_object:
+        seen.sobject = 1
 
+    if not seen.sobject:
         if not safe:
```

Comment:
No actually because when we detect conflicts, `seen.sobject` is set to 1, and so that whole branch of logic is skipped.